### PR TITLE
Make CMake script macOS-compatible

### PR DESCRIPTION
**Pull request by OPNA2608**
Closes https://github.com/8bitbubsy/ft2-clone/issues/13

### CODE REVIEW

**Comment by 8bitbubsy** [comment]
1) Does it work on both Intel Mac and ARM mac?
2) __MAXOSX_CORE__?

**Comment by OPNA2608** [comment]
1. I only have access to an Intel Mac, and it seems to work fine there. (I'm seeing some problems with the program on it but I don't think they're related to these settings)
2. [Needed by RtMidi to enable CoreMIDI support](https://github.com/8bitbubsy/ft2-clone/blob/9cf2fd9416bc0e3586e191fb732c7e7903bdce98/src/rtmidi/RtMidi.cpp#L64-L103)

**Comment by 8bitbubsy** [comment]
1. I'm not going to add this if it doesn't compile on ARM Macs, I need someone with an ARM Mac to verify.
2. Let me try again. :) __MAXOSX_CORE__? __MACOSX_CORE__!

**Comment by OPNA2608** [comment]
Whoops! :sweat_smile:

**Comment by h57758** [comment]
I have an M1 Mac Mini we could test on!

**Comment by OPNA2608** [comment]
> I'm not going to add this if it doesn't compile on ARM Macs

> @OPNA2608 We could try with the nixos builders

Tested on aarch64-darwin via Nixpkgs CI test infrastructure: https://logs.ofborg.org/?attempt_id=663979be-7320-42cc-8225-1daf801410e7&key=nixos%2Fnixpkgs.250732 , [aarch64-darwin_ft2-clone_cmake-patch.log.txt](https://github.com/8bitbubsy/ft2-clone/files/12409296/aarch64-darwin_ft2-clone_cmake-patch.log.txt)

**Comment by fgaz** [comment]
@8bitbubsy ping! This patch has been working well in our builds https://hydra.nixos.org/job/nixpkgs/trunk/ft2-clone.aarch64-darwin

**Comment by 8bitbubsy** [comment]
I fixed CMakeLists.txt in the latest commit. Please let me know if it works @fgaz and @OPNA2608.

**Comment by fgaz** [comment]
It works, thank you!

https://github.com/NixOS/nixpkgs/pull/259155

**Comment by OPNA2608** [comment]
The aarch64 build works on my PineBook Pro running NixOS FWIW.
